languages/csharp/reference/README.md

```diff
@@ -8,89 +8,136 @@ The C# concept exercises are based on concepts. The list below contains the conc
-
-- [Assemblies](../../../reference/tooling/dotnet-assemblies.md)
-
-### Memory management
-
-- Resource passing
-- [Resource allocation](./memory_allocation.md) (can include `Span<T>`/`Memory<T>` types)
-- Resource cleanup
-- Resource lifetime
-
-### Methods
-
-- Method overloading
-- Named arguments
-- Optional arguments
-- Extension methods (mixin)
+- [Functions][functions]
+  - [Higher-order functions][higher_order_functions]
+  - [Anonymous functions][anonymous_functions]
+  - [Local functions][local_functions]
+- [Immutability][immutability]
+- [Pattern matching][pattern_matching]
+- [Type inference][type_inference]
+- [Recursion][recursion]
+- [LINQ][linq]
+- Expressions vs statements
 
 ### General
 
-- Arithmetic overflow
-- [Sameness](../../../reference/concepts/sameness.md)
-- [Conditionals](../../../reference/concepts/conditionals.md)
-- [Enumeration](../../../reference/concepts/enumeration.md)
-- Iterators (yield)
-- Namespaces
-- [Generics](../../../reference/concepts/generics.md)
-- Exception handling
-- Implicit/explicit conversion
-- Boxing/unboxing
-- Anonymous types
-- Concurrency (can include concurrent collections)
+- [Variables]
+  - Assignment
+  - Default values (a `bool` being `false` by default, etc.)
+- Scoping
+  - Namespaces
+  - Imports (usings)
+  - Visibility (`public`, `private`, etc.)
+- Numbers
+  - Math operators
+  - Bitwise manipulation
+  - Arithmetic overflow
+- Comparison
+  - [Equality][equality] (`Equals`, `GetHashCode`)
+  - Ordering
+- Conditionals
+  - Boolean logic
+  - [Conditionals: if][conditionals]
+  - [Conditionals: ternary][conditionals]
+  - Conditionals: while
+  - Conditionals: do-while
+  - Conditionals: switch
+- Enumeration
+  - [Enumeration: for loop][enumeration]
+  - [Enumeration: foreach loop][enumeration]
+- Collections
+  - [Generics][generics]
+  - Iterators (yield)
+  - Collections: mapping
+  - Collections: filtering
+  - Collections: reducing
+  - Collections: ordering
+  - Collections: combining
+- Resources
+  - Resource passing (by reference/by value)
+  - [Resource allocation][memory_allocation]
+  - Resource cleanup (`IDisposable`)
+  - Resource lifetime
+- Conversions
+  - Implicit
+  - Explicit (casts)
+  - Boxing/unboxing
+- Null
+  - Nullable values
+  - Null-coalescing operator
+  - Null-conditional operator
+- Concurrency
+  - Concurrent collections
+  - Locks
+- Exceptions
 - Asynchronous programming
-- Expressions vs statements
 - Attributes
 - Slicing
 - Unsafe code
-- Reflection
-- Nullability
+- Comments
+- Randomness
+- Regular expressions
+- String formatting
+  - Interpolation
+  - Formatting types
 
 ### Types
 
-- Signed integers
-- Unsigned integers
-- Floating point numbers
 - [Strings][string]
-- Enums
 - [Characters][char]
 - [Booleans][bool]
-- [Arrays][array]
-- [Lists][list]
-- [Dictionaries][map]
-- [Sets][set]
-- [Queues][queue]
-- [Stacks][stack]
+- Enums
 - [Tuples][tuple]
-- [Ranges][range]
+- Numbers
+  - Signed integers
+  - Unsigned integers
```

Comment:
- Decimals

Reply:
According to the [official docs](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/floating-point-numeric-types), decimals are considered a floating-point number, and as such are covered in the next "Floating point numbers" Concept.

Reply:
Sounds good